src/main/java/com/github/princesslana/smalld/ratelimit/RateLimitInterceptor.java

```diff
@@ -113,17 +113,17 @@ private void setRateLimitForPath(Request request, RateLimit rateLimit) {
             .map(Instant::from);
 
     Optional<Instant> retryAfter =
-        getRetryAfter(response).map(responseDate.orElse(clock.instant())::plusMillis);
+        getRetryAfter(response).map(responseDate.orElse(clock.instant())::plusSeconds);
 
     return Stream.of(reset, retryAfter).filter(Optional::isPresent).map(Optional::get).findFirst();
   }
 
-  private Optional<Long> getRetryAfter(Response response) {
-    return headerAsLong(response, "Retry-After");
+  private Optional<Integer> getRetryAfter(Response response) {
```

Comment:
Why change these to Integers?

Reply:
Yup, now I see that was useless i will change it to Long again.